From 2b6bf935992d2209984c6ca1cf11cf548ced5560 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Thu, 4 Feb 2021 20:44:30 +0100
Subject: [PATCH] Fix static linking for libopenjpeg >= 2.4

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a778bbd7..a97a2d00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,6 +259,9 @@ else()
   message(FATAL_ERROR "Invalid ENABLE_LIBOPENJPEG value: ${ENABLE_LIBOPENJPEG}")
 endif()
 set(ENABLE_LIBOPENJPEG "${WITH_OPENJPEG}")
+if(ENABLE_LIBOPENJPEG AND NOT BUILD_SHARED_LIBS)
+  add_definitions(-DOPJ_STATIC)
+endif()
 if(ENABLE_CMS STREQUAL "lcms2")
   find_package(LCMS2)
   set(USE_CMS ${LCMS2_FOUND})
-- 
2.27.0

