--- a/qt5/lib/cmake/Qt5UiPlugin/Qt5UiPluginConfig.cmake	2016-02-28 09:49:50.814090672 -0500
+++ b/qt5/lib/cmake/Qt5UiPlugin/Qt5UiPluginConfig.cmake	2016-02-07 11:35:46.878858995 -0500
@@ -88,8 +88,7 @@
     set(_Qt5UiPlugin_LIB_DEPENDENCIES "Qt5::Widgets;Qt5::Gui;Qt5::Core")
 
 
-    add_library(Qt5::UiPlugin STATIC IMPORTED)
-    set_property(TARGET Qt5::UiPlugin PROPERTY IMPORTED_LINK_INTERFACE_LANGUAGES CXX)
+    add_library(Qt5::UiPlugin INTERFACE IMPORTED)
 
     set_property(TARGET Qt5::UiPlugin PROPERTY
       INTERFACE_INCLUDE_DIRECTORIES ${_Qt5UiPlugin_OWN_INCLUDE_DIRS})

