diff --git a/icu4c/source/data/makedata.mak b/icu4c/source/data/makedata.mak
index 1a48ee0..d573f23 100644
--- a/icu4c/source/data/makedata.mak
+++ b/icu4c/source/data/makedata.mak
@@ -481,16 +481,17 @@ GODATA : CREATE_DIRS
 CLEAN : GODATA
 	@echo Cleaning up the data files.
 	@cd "$(ICUOUT)"
+	dir
 	-@erase "$(ICUOUT)\*.dat"
-	@rmdir $(ICUBLD) /s /q
-	@rmdir $(ICUTMP) /s /q
-	@rmdir $(TESTDATAOUT) /s /q
+	-erase /f /s /q /a $(ICUBLD)\*
+	-erase /f /s /q /a $(ICUTMP)\*
+	-erase /f /s /q /a $(TESTDATAOUT)\*
 
 # DLL version information
 # If you modify this, modify winmode.c in pkgdata.
 "$(ICUTMP)\icudata.res": "$(ICUSRCDATA)\misc\icudata.rc"
 	@echo Creating data DLL version information from $**
-	@rc.exe /i "..\..\..\..\common" /r /fo $@ $**
+	rc.exe /i "$(ICUSRCDATA)\..\common" /r /fo $@ $**
 
 # Targets for prebuilt Unicode data
 # Needed for ICU4J!
diff --git a/icu4c/source/install.cmake b/icu4c/source/install.cmake
new file mode 100644
index 0000000..5fd7934
--- /dev/null
+++ b/icu4c/source/install.cmake
@@ -0,0 +1,14 @@
+if (${PLATFORM} STREQUAL "ARM64")
+	set(ICU_LIBDIR "libARM64")
+	set(ICU_BINDIR "binARM64")
+else ()
+	set(ICU_LIBDIR "lib64")
+	set(ICU_BINDIR "bin64")
+endif ()
+
+file(GLOB LIBS icu4c/${ICU_LIBDIR}/*.lib)
+file(GLOB DLLS icu4c/${ICU_BINDIR}/*.dll)
+
+file(INSTALL ${LIBS} DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+file(INSTALL ${DLLS} DESTINATION ${CMAKE_INSTALL_FULL_BINDIR})
+file(INSTALL icu4c/include/unicode DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/)
