From 05c33c34e6cb5624092457c09e6041c01d5933a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20L=C3=B6ffler?= <st.loeffler@gmail.com>
Date: Tue, 25 Jan 2022 22:47:29 +0100
Subject: [PATCH] Fix building with MSVC and libjpeg

---
 poppler/ImageEmbeddingUtils.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/poppler/ImageEmbeddingUtils.cc b/poppler/ImageEmbeddingUtils.cc
index cd7a086..a7f529a 100644
--- a/poppler/ImageEmbeddingUtils.cc
+++ b/poppler/ImageEmbeddingUtils.cc
@@ -17,6 +17,7 @@
 #include <memory>
 #if ENABLE_LIBJPEG
 #    include <cstdio> // jpeglib.h is broken on purpose and needs cstdio include before it
+#    include <basetsd.h>
 extern "C" {
 #    include <jpeglib.h>
 }
-- 
2.51.0

