diff --git debian/control debian/control
index 5c59c0b..623d571 100644
--- debian/control
+++ debian/control
@@ -2,7 +2,7 @@ Source: texworks
 Section: tex
 Priority: optional
 Maintainer: Stefan Löffler <st.loeffler@gmail.com>
-Build-Depends: debhelper (>= 9), pkg-config, cmake (>= 3.1), qt6-5compat-dev, qt6-base-dev, qt6-declarative-dev, qt6-tools-dev, zlib1g-dev, libhunspell-dev, libpoppler-qt6-dev, libpoppler-private-dev, liblua5.4-dev, xvfb, fonts-urw-base35, poppler-data
+Build-Depends: debhelper (>= 9), pkg-config, cmake (>= 3.1), qtbase5-dev, qtdeclarative5-dev, qtscript5-dev, qttools5-dev, qttools5-dev-tools, zlib1g-dev, libhunspell-dev, libpoppler-qt5-dev, libpoppler-private-dev, liblua5.3-dev, xvfb, gsfonts, poppler-data
 Standards-Version: 3.9.7
 Homepage: https://github.com/TeXworks/texworks
 Vcs-Git: https://github.com/TeXworks/texworks.git
diff --git debian/rules debian/rules
index fafb736..f7116f2 100755
--- debian/rules
+++ debian/rules
@@ -16,7 +16,7 @@ configure: configure-stamp
 configure-stamp:
 	dh_testdir
 	# Configure
-	mkdir --parents build && cd build && cmake .. -DTW_BUILD_ID='launchpad' -DCMAKE_INSTALL_PREFIX='/usr' -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DQT_DEFAULT_MAJOR_VERSION=6 && cd ..
+	mkdir --parents build && cd build && cmake .. -DTW_BUILD_ID='launchpad' -DCMAKE_INSTALL_PREFIX='/usr' -DCMAKE_BUILD_TYPE="RelWithDebInfo" && cd ..
 	touch $@
 
 # Build architecture-independent files
